File: lab2/deserialize_json.py
import json


class DeserializeJson:
    def __init__(self, filename):
        print("let's deserialize something")
        with open(filename, encoding="utf8") as f:
            self.data = json.load(f)

    def somestats(self):
        example_stat = 0

        departments_in_voivodeship_count = {}

        for department in self.data:
            if department["typ_JST"] == "GM" and department["Województwo"] == "dolnośląskie":
                example_stat += 1

            if department["Województwo"] not in departments_in_voivodeship_count:
                departments_in_voivodeship_count[department["Województwo"]] = 1
            else:
                departments_in_voivodeship_count[department["Województwo"]] += 1

        print(f"liczba urzędów miejskich w województwie dolnośląskim: {example_stat}")
        print("Liczba urzędów w wojewodztwach: ")
        for voivodeship, count in departments_in_voivodeship_count.items():
            print(f"\t{voivodeship}: {count}")

File: lab2task6/lab2task6/lab2task6/DeserializeJson.cs
using System;
using System.Collections.Generic;
using System.IO;
using Newtonsoft.Json;

namespace lab2task6
{
    public class DeserializeJson
    {
        public dynamic Data { get; }

        public DeserializeJson(string sourcePath)
        {
            using var reader = new StreamReader(sourcePath);
            var json = reader.ReadToEnd();
            Data = JsonConvert.DeserializeObject(json);
        }

        public void SomeStats()
        {
            var exampleStat = 0;
            var departmentsInVoivodeshipCount = new Dictionary<string, int>();
            foreach (var department in Data)
            {
                string voivodeship = department["Województwo"];
                
                if (department["typ_JST"] == "GM" && voivodeship == "dolnośląskie")
                    exampleStat++;

                if (!departmentsInVoivodeshipCount.ContainsKey(voivodeship))
                    departmentsInVoivodeshipCount[voivodeship] = 1;
                else
                    departmentsInVoivodeshipCount[voivodeship]++;
            }

            Console.WriteLine($"liczba urzędów miejskich w województwie dolnośląskim: {exampleStat}");
            Console.WriteLine("Liczba urzędów w wojewodztwach: ");
            foreach (var (voivodeship, count) in departmentsInVoivodeshipCount)
                Console.WriteLine($"\t{voivodeship}: {count}");
        }
    }
}